simpeg_drivers/utils/regularization.py

```diff
@@ -26,7 +26,11 @@ def cell_neighbors_along_axis(mesh: TreeMesh, axis: str) -> np.ndarray:
     if axis not in "xyz":
         raise ValueError("Argument 'axis' must be one of 'x', 'y', or 'z'.")
 
-    stencil = getattr(mesh, f"cell_gradient_{axis}")
+    if isinstance(mesh, TreeMesh):
+        stencil = getattr(mesh, f"stencil_cell_gradient_{axis}")
```

Comment:
why do we need this logic here, but not at line 482 in the driver?

Reply:
Because here mesh is the `discretize` object (either TreeMesh or Tensor). At line 482, we are overloading the reg_mesh._cell_gradient_ directly

Reply:
k